src/codegen/statements.rs

```diff
@@ -63,21 +63,76 @@ pub(crate) fn statement(
                     vartab,
                     opt,
                 };
-
                 //If we remove the assignment, we must keep expressions that have side effects
                 init.recurse(&mut params, process_side_effects_expressions);
                 return;
             }
 
             let expr = expression(init, cfg, contract_no, Some(func), ns, vartab, opt);
+
             cfg.add(
                 vartab,
                 Instr::Set {
                     loc: *loc,
                     res: *pos,
-                    expr,
+                    expr: expr.clone(),
                 },
             );
+
+            // Lets check if the declaration is a declaration of a dynamic array
+            if let Expression::AllocDynamicArray(_loc, _ty, size, _option) = expr {
+                // If its a variable of uint32, assign it to a temp
+                if let Expression::Variable(_loc, ref ty, position) = *size {
```

Comment:
Is it necessary to dereference `size` here?

Reply:
I think yes, because here size is of type Box<Expression>